apps/frontend/src/components/ChessBoard.tsx

```diff
@@ -186,8 +186,17 @@ export const ChessBoard = ({
                         }
                         if (!from && square?.color !== chess.turn()) return;
                         if (!isMyTurn) return;
+
                         if (from === squareRepresentation) {
                           setFrom(null);
+                        } else {
+                          setFrom(squareRepresentation);
+
+                          setLegalMoves(
```

Comment:
Could u confirm this works if u move ur peice?
It seems like we override from even when the user might be moving somewhere where their piece isnt present

Reply:
https://github.com/code100x/chess/assets/136328410/b4420e28-0e5a-47ae-8177-1db0e1f1cdf9

you can move the piece i even added a feature where if you click anywhere else on the board it deselects 
there was a small bug of the board displaying invalid moves but i fixed that as well you can review it